app/lib/shared/handlers.dart

```diff
@@ -101,16 +104,21 @@ shelf.Response debugResponse([Map<String, dynamic> data]) {
 }
 
 bool isNotModified(shelf.Request request, DateTime lastModified, String etag) {
-  final ifModifiedSince = request.ifModifiedSince;
-  if (ifModifiedSince != null &&
-      lastModified != null &&
-      !lastModified.isAfter(ifModifiedSince)) {
-    return true;
-  }
-
-  final ifNoneMatch = request.headers[HttpHeaders.ifNoneMatchHeader];
-  if (ifNoneMatch != null && ifNoneMatch == etag) {
-    return true;
+  try {
+    final ifModifiedSince = request.ifModifiedSince;
+    if (ifModifiedSince != null &&
+        lastModified != null &&
+        !lastModified.isAfter(ifModifiedSince)) {
+      return true;
+    }
+
+    final ifNoneMatch = request.headers[HttpHeaders.ifNoneMatchHeader];
+    if (ifNoneMatch != null && ifNoneMatch == etag) {
+      return true;
+    }
+  } catch (e, st) {
```

Comment:
Can we catch an exact exception? Note: If this is a client error, we should fail 400 rather than 500.